tests/test_connector.py

```diff
@@ -0,0 +1,66 @@
+from logging import getLogger
+from unittest import mock
+
+from pytest import fixture, fail
+
+import code42_connector
+from phantom.action_result import ActionResult
+import phantom.app as phantom
+import py42.sdk
+from requests import HTTPError
+
+
+logger = getLogger(name=__name__)
+
+
+NULL_VALUE = object()
+
+
+@fixture
+def py42_sdk(mocker):
+    return mocker.MagicMock(spec=py42.sdk)
+
+
+@fixture
+def py42_client(mocker):
+    return mocker.MagicMock(spec=py42.sdk.SDKClient)
+
+
+@fixture
+def connector(request, py42_client):
+    connector = code42_connector.Code42Connector()
+    connector._client = py42_client
+    return connector
+
+
+def test_handle_connectivity(mocker, connector):
+    param = {}
+    mocker.patch('phantom.base_connector.BaseConnector.add_action_result', return_value=ActionResult(dict(param)))
+    set_status = mocker.patch('phantom.action_result.ActionResult.set_status')
+
+    connector._handle_test_connectivity(param)
+
+    set_status.assert_called_with(phantom.APP_SUCCESS)
+
+
+def test_handle_connectivity_sets_error_status_if_sdk_throws_exception(mocker, connector, py42_client):
+    param = {}
+    mocker.patch('phantom.base_connector.BaseConnector.add_action_result', return_value=ActionResult(dict(param)))
+    py42_client.users.get_current.side_effect = HTTPError(mock.Mock(status=401))
+    set_status = mocker.patch('phantom.action_result.ActionResult.set_status')
+
+    connector._handle_test_connectivity(param)
+
+    set_status.assert_called_with(phantom.APP_ERROR, mocker.ANY, mocker.ANY)
+
+
+def test_handle_action_calls_handle_connectivity(mocker, connector):
+    param = {}
+    handle_test_connectivity = mocker.patch('code42_connector.Code42Connector._handle_test_connectivity',
```

Comment:
In general, we should avoid mocking internal methods and instead treat them as implementation details.

Ideally, the tests above should probably be testing `handle_action` instead of `handle_test_connectivity` but still test the the same functionality.

At that point, I don't think this test would even be needed

Reply:
instead of patching, you could also create a utility (note: not fixture) method that would produce a version of the connector with a mocked `get_action_identifier`:

```python
def create_fake_connector(action_identifier):

    def fake_get_action_idenfifer():
        return action_identifier
        
    connector = code42_connector.Code42Connector()
    connector.get_action_identifier = fake_get_action_identifier
    return connector
```

Then we could do what Jules is suggesting by forcing tests to take the right path:
```python
@fixture
def test_connectivity_connector(py42_client):
   connector = create_fake_connector("test_connectivity")
   connector._client = py42_client
   return connector
```
then `test_connectivity_connector.handle_action()` should automatically serve as a test that the correct internal method is called (whatever output or other effects we're testing would presumably fail if that _werent_ the case.

Reply:
Implemented Alan's suggestion